### PR TITLE
ci: add condition to have the same pipeline in all branches

### PR DESCRIPTION
**Pull request by kuisathaverat**
it adds `when` condition to have the same pipeline in all branches.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -88,6 +90,45 @@ pipeline {
         }
       }
     }
+    /**
+     Publish Docker images.
+     */
+    stage('Publish Docker image'){
+      when {
+        allOf {
+          anyOf {
```
Comment (kuisathaverat) [inline .ci/Jenkinsfile:99]: this filter which branches create a `docker.elastic.co/package-registry/distribution` Docker image

**Comment by kuisathaverat** [comment]
there is a lint issue on the master branch, so the error is not related to my changes.

**Comment by ruflin** [comment]
@kuisathaverat We are discussing to potentially remove the master branch :-) I assume the reason you put it into master is to make it reusable?

**Comment by kuisathaverat** [comment]
>I assume the reason you put it into master is to make it reusable?

not really, it is there because the master branch used to be the main branch :D 

>We are discussing to potentially remove the master branch 

we need the master branch, it is where the jobs' definition is

**Comment by ruflin** [comment]
@kuisathaverat I assume we don't directly need the master branch but a "predefined" branch with all the definitions?

**Comment by kuisathaverat** [comment]
@ruflin I have to check JJBB, but I think we can point it to any branch but it is not the standard way.

**Comment by ruflin** [comment]
Ok, I'll ping you in case we rename / delete the master branch.

**Pull request merged**